client/my-sites/media-library/list-item.tsx

```diff
@@ -0,0 +1,159 @@
+import EditorMediaModalGalleryHelp from 'post-editor/media-modal/gallery-help';
+
+/**
+ * Style dependencies
+ */
+import './list-item.scss';
+
+// TODO: move to lib/media/utils once it gets typed.
+interface MediaObject {
+	transient?: boolean;
+	file?: string;
+	[propName: string]: any;
+}
+type Media = string | MediaObject;
+// END TODO
+
+interface Props {
+	media?: Media;
+	scale: number;
+	maxImageWidth?: number;
+	thumbnailType?: string;
+	showGalleryHelp?: boolean;
+	selectedIndex?: number;
+	onToggle?: ( media: Media | undefined, shiftKey: boolean ) => any;
+	onEditItem?: any; // Unused. Appears to have been left here for compatibility reasons.
+	style?: React.CSSProperties;
+}
+
+type DivProps = Omit< React.ComponentProps< 'div' >, 'style' | 'onClick' >;
+
+export default class MediaLibraryListItem extends React.Component< Props & DivProps > {
+	static defaultProps = {
+		maxImageWidth: 450,
+		selectedIndex: -1,
+		onToggle: noop,
+		onEditItem: noop,
+	};
+
+	shouldComponentUpdate( nextProps: Props ) {
+		return ! (
+			nextProps.media === this.props.media &&
+			nextProps.scale === this.props.scale &&
+			nextProps.maxImageWidth === this.props.maxImageWidth &&
+			nextProps.thumbnailType === this.props.thumbnailType &&
+			nextProps.selectedIndex === this.props.selectedIndex &&
+			isEqual( nextProps.style, this.props.style )
+		);
+	}
+
+	clickItem = ( event: React.MouseEvent ) => {
+		if ( this.props.onToggle ) {
+			this.props.onToggle( this.props.media, event.shiftKey );
+		}
+	};
+
+	renderItem = () => {
+		let component;
+
+		if ( ! this.props.media ) {
+			return;
+		}
+
+		switch ( getMimePrefix( this.props.media ) as string ) {
+			case 'image':
+				component = ListItemImage;
+				break;
+			case 'video':
+				component = ListItemVideo;
+				break;
+			case 'audio':
+				component = ListItemAudio;
+				break;
+			default:
+				component = ListItemDocument;
+				break;
+		}
+
+		return React.createElement( component, this.props );
+	};
+
+	render() {
+		let title, selectedNumber;
+
+		const {
+			media,
+			scale,
+			maxImageWidth,
+			thumbnailType,
+			showGalleryHelp,
+			selectedIndex,
+			onToggle,
+			onEditItem,
+			style,
+			...otherProps
+		} = this.props;
+
+		let dataAttributes = null;
+
+		const classes = classNames( 'media-library__list-item', {
+			'is-placeholder': ! media,
+			'is-selected': -1 !== selectedIndex,
+			'is-transient': media && ( media as MediaObject ).transient,
+			'is-small': scale <= 0.125,
+		} );
+
+		const styleWithDefaults = { width: scale * 100 + '%', ...style };
+
+		if ( media && ( media as MediaObject ).file ) {
+			title = ( media as MediaObject ).file;
+		}
+
+		if ( selectedIndex !== -1 && selectedIndex !== undefined ) {
+			selectedNumber = selectedIndex + 1;
+			dataAttributes = { 'data-selected-number': selectedNumber > 99 ? '99+' : selectedNumber };
+		}
+
+		return (
+			/* eslint-disable jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions */
```

Comment:
It would be nice to give the `div` a reasonable a11y role to prevent this warning. Or use a different element, e.g., `button`.

Reply:
Switched to `<button>` and it doesn't seem to break styling.